atd-vze/src/views/Users/Users.js

```diff
@@ -83,8 +87,57 @@ const Users = () => {
       });
   }, [token, page, perPage]);
 
+  useEffect(() => {
+    // if the user clicks the copy users button and we have received user metadata,
+    // then query all users
+    !!copyUserEmailsClicked &&
+      !!totalUsers &&
+      // only do this if the user hasn't already fetched data. if they have, skip this step
+      // and copy what is in state to the clipboard
+      allUsers?.length !== totalUsers &&
+      getAllUsers();
+    // once we have all users, copy the emails to the clipboard
+    !!allUsers.length && !!copyUserEmailsClicked && getUserEmails();
+  });
```

Comment:
I was expecting to see a dependency array in this `useEffect` but I see that your boolean conditions isolate how often this code fires so I think this is a-okay. 👍

Reply:
will netlify still build this without a dependency array?

Reply:
yes, i think the only warning that we would see that might affect the build is if there was a dependency array that was missing dependencies. Without the array, React just treats this as "run this on every render" and the boolean checks prevent any code beyond them from running.

Reply:
I feel a bit more strongly that there needs to have a dependency array. if absolutely necessary, i would prefer to see the array and an eslint ignore rather than no array at all. with no array, the hook runs every render and so there's not really any point to having it at all. 

if putting the api call in the useEffect hook is feeling awkward, it might be worth trying to refactor so that the API call function is fired directly by the button click event. 

it also seems like the two chunks of code that call the `list_users` API could be combined if they do the same thing but for paginating.